assets/blocks/quiz/category-question-block/category-question-settings.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * WordPress dependencies
+ */
+import { InspectorControls } from '@wordpress/block-editor';
+import { Notice, PanelBody, SelectControl } from '@wordpress/components';
+import { __, _n, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import NumberControl from '../../editor-components/number-control';
+import { useQuestionCategories } from '../question-categories';
+
+/**
+ * Category question block settings controls.
+ *
+ * @param {Object}   props                    Block props.
+ * @param {Object}   props.attributes         Block attributes.
+ * @param {Object}   props.attributes.options Block options attribute.
+ * @param {Function} props.setAttributes      Update block attributes.
+ */
+const CategoryQuestionSettings = ( {
+	attributes: { options = {} },
+	setAttributes,
+} ) => {
+	const setOptions = ( next, otherAttributes = {} ) =>
+		setAttributes( {
+			...otherAttributes,
+			options: { ...options, ...next },
+		} );
+
+	const [
+		questionCategories,
+		getQuestionCategoryById,
+	] = useQuestionCategories();
+
+	const categoryOptions = [
+		{
+			value: '',
+			label: __( 'Category', 'sensei-lms' ),
+		},
+		...( questionCategories || [] ).map( ( questionCategory ) => ( {
+			value: questionCategory.id,
+			label: questionCategory.name,
+		} ) ),
+	];
+
+	const questionCategory = getQuestionCategoryById( options.category );
+
+	return (
+		<InspectorControls>
+			<PanelBody
+				title={ __( 'Category Question Settings', 'sensei-lms' ) }
+				initialOpen={ true }
+			>
+				{ ! categoryOptions.length && (
+					<Notice status="warning" isDismissible={ false }>
+						{ __( 'No question categories exist.', 'sensei-lms' ) }
+					</Notice>
+				) }
+				{ categoryOptions.length > 0 && (
+					<>
+						<SelectControl
+							label={ __( 'Category', 'sensei-lms' ) }
+							options={ categoryOptions }
+							value={ options.category }
```

Comment:
We need a check to set an empty string when it's `null` to avoid a React error.

Reply:
Done in 307656c1e2b3428f6b2cc1df6e09b770e7163f6e